xbmc/cores/RetroPlayer/rendering/RPRenderManager.cpp

```diff
@@ -918,3 +974,72 @@ void CRPRenderManager::FreeVideoFrame(IRenderBuffer* readableBuffer,
     m_cachedFrame = std::move(cachedFrame);
   }
 }
+
+void CRPRenderManager::LoadVideoFrameAsync(const std::string& savestatePath)
+{
+  // Prune any finished loader threads
+  m_savestateThreads.erase(std::remove_if(m_savestateThreads.begin(), m_savestateThreads.end(),
+                                          [](std::future<void>& task) {
+                                            return task.wait_for(std::chrono::seconds(0)) ==
+                                                   std::future_status::ready;
+                                          }),
+                           m_savestateThreads.end());
+
+  // Load the video data from the savestate asynchronously
+  std::future<void> task = std::async(
+      std::launch::async, [this, savestatePath]() { LoadVideoFrameSync(savestatePath); });
+
+  m_savestateThreads.emplace_back(std::move(task));
+}
+
+void CRPRenderManager::LoadVideoFrameSync(const std::string& savestatePath)
+{
+  if (!XFILE::CFile::Exists(savestatePath))
+  {
+    CLog::Log(LOGERROR, "Failed to load savestate: doesn't exist at path {}",
+              CURL::GetRedacted(savestatePath));
+    return;
+  }
+
+  std::unique_ptr<ISavestate> savestate = CSavestateDatabase::AllocateSavestate();
+  CSavestateDatabase db;
+  if (!db.GetSavestate(savestatePath, *savestate))
+    return;
+
+  // Load video data
+  const AVPixelFormat format = savestate->GetPixelFormat();
+  const uint8_t* data = savestate->GetVideoData();
+  const size_t size = savestate->GetVideoSize();
+  const unsigned int width = savestate->GetVideoWidth();
+  const unsigned int height = savestate->GetVideoHeight();
+  const unsigned int rotationCCW = savestate->GetRotationDegCCW();
+
+  // Validate parameters
+  if (format == AV_PIX_FMT_NONE || data == nullptr || size == 0 || width == 0 || height == 0)
+  {
+    CLog::Log(LOGERROR, "Invalid video data: format {}, data {}, size {}, width {}, height {}",
+              static_cast<long int>(format), static_cast<const void*>(data), size, width, height);
+    return;
+  }
+
+  // Copy frame to render buffers
+  std::vector<IRenderBuffer*> renderBuffers;
+  for (auto* bufferPool : m_processInfo.GetBufferManager().GetBufferPools())
```

Comment:
Should the `*` be removed? The auto should resolve to `IRenderBufferPool*` or are you just being explicit?

Reply:
That was just what popped into my head. `auto` is preferred. But now that I look at it, I kind of like the extra expliciticity.

Reply:
not to mention dat word